services/proxy/pkg/router/router.go

```diff
@@ -207,32 +209,27 @@ func (rt Router) Route(r *http.Request) (RoutingInfo, bool) {
 			method = r.Method
 		}
 
-		endpoints := make([]string, 0, len(rt.directors[pol][rtype][method]))
-		for endpoint := range rt.directors[pol][rtype][method] {
-			endpoints = append(endpoints, endpoint)
-		}
-		sort.Slice(endpoints, func(i, j int) bool {
-			return len(endpoints[j]) < len(endpoints[i])
-		})
-		for _, endpoint := range endpoints {
-			if handler(endpoint, *r.URL) {
+		for _, ri := range rt.directors[pol][rtype][method] {
+			rt.logger.Error().Str("endpoint", ri.endpoint).Msg("testing")
+			if handler(ri.endpoint, *r.URL) {
 				rt.logger.Debug().
 					Str("policy", pol).
 					Str("method", r.Method).
-					Str("prefix", endpoint).
+					Str("prefix", ri.endpoint).
 					Str("path", r.URL.Path).
 					Str("routeType", string(rtype)).
 					Msg("director found")
 
-				return rt.directors[pol][rtype][method][endpoint], true
+				rt.logger.Error().Str("endpoint", ri.endpoint).Msg("match")
+				return ri, true
 			}
 		}
 	}
 
 	// override default director with root. If any
-	if ri, ok := rt.directors[pol][config.PrefixRoute][method]["/"]; ok { // try specific method
+	if ri := rt.directors[pol][config.PrefixRoute][method][0]; ri.endpoint == "/" { // try specific method
```

Comment:
I am still not too happy with this. I guess I need something better here for the fallback/default routes.